routers/web/repo/pull.go

```diff
@@ -395,12 +395,17 @@ func PrepareViewPullInfo(ctx *context.Context, issue *issues_model.Issue) *git.C
 	var headBranchSha string
 	// HeadRepo may be missing
 	if pull.HeadRepo != nil {
-		headGitRepo, err := gitrepo.OpenRepository(ctx, pull.HeadRepo)
-		if err != nil {
-			ctx.ServerError("OpenRepository", err)
-			return nil
+		var headGitRepo *git.Repository
+		if ctx.Repo != nil && ctx.Repo.Repository != nil && pull.HeadRepoID == ctx.Repo.Repository.ID && ctx.Repo.GitRepo != nil {
```

Comment:
If I understand correctly, `ctx.Repo.Repository` can't be `nil` here

Reply:
Fixed in https://github.com/go-gitea/gitea/pull/32393/commits/3460918c704ac4562c682ac78a265a7b2e1cb476